extensions/cl_ext_cxx_for_opencl.asciidoc

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+= cl_ext_cxx_for_opencl
+:R: pass:q,r[^(R)^]
+Khronos{R} OpenCL Working Group
+
+== Name Strings
+
+`cl_ext_cxx_for_opencl`
+
+== Contact
+
+Please see the *Issues* list in the Khronos *OpenCL-Docs* repository: +
+https://github.com/KhronosGroup/OpenCL-Docs
+
+== Contributors
+
+Kevin Petit, Arm Ltd. +
+Sven Van Haastregt, Arm Ltd. +
+Anastasia Stulova, Arm Ltd. +
+Marco Antognini, Arm Ltd. +
+Neil Hickey, Arm Ltd. +
+
+== Notice
+
+include::../copyrights.txt[]
+
+== Version
+
+Built On: {docdate} +
+Version: 1.0.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 3.0.2.
+
+This extension requires OpenCL 3.0 or OpenCL 2.x and `cl_khr_extended_versioning`.
+
+== Overview
+
+This extension adds support for building programs written using the C++ for
+OpenCL kernel language. It also enables applications to query the version of
+the language supported by the device compiler.
+
+== New build option
+
+This extension adds support for a new `CLC++` value to be passed to the
+`-cl-std` build option accepted by *clBuildProgram* and *clCompileProgram*.
+
+This option is only valid for programs created using
+*clCreateProgramFromSource*.
```

Comment:
The definition of `clCreateProgramFromSource` says:

> The devices associated with the program object are the devices associated with context. The source code specified by strings is either an OpenCL C program source, header or implementation-defined source for custom devices that support an online compiler. OpenCL C++ is not supported as an online-compiled kernel language through this interface.

Should this extension say something about causing `clCreateProgramFromSource` to accept C++ for OpenCL source?

Reply:
This is implied by the current text but being more explicit wouldn't hurt. I'll add something.

Reply:
Yeah, I definitely agree that it's implied.